src/app/inventory/selectors.ts

```diff
@@ -106,25 +109,83 @@ export const materialsSelector = (state: RootState) =>
 /** The actual raw profile response from the Bungie.net profile API */
 export const profileResponseSelector = (state: RootState) => state.inventory.profileResponse;
 
+const STORE_SPECIFIC_OWNERSHIP_BUCKETS = [
+  // Emblems cannot be transferred between characters and if one character owns an emblem,
+  // other characters don't really own it. Also affects vendor claimability.
+  BucketHashes.Emblems,
+  // Quests and bounties are character-specific.
+  BucketHashes.Quests,
+];
+
+/**
+ * For most items, ownership is simple -- if it's in any bucket, we own this item.
+ * For items in some buckets (like quests and emblems), there is no concept of
+ * account-wide ownership. This mostly affects vendor claimability.
+ */
+export interface OwnedItems {
+  accountWideOwned: Set<number>;
+  storeSpecificOwned: Map<string, Set<number>>;
+}
+
 /** A set containing all the hashes of owned items. */
 export const ownedItemsSelector = createSelector(
+  (state: RootState) => state.manifest.d2Manifest,
   profileResponseSelector,
   allItemsSelector,
-  (profileResponse, allItems) => {
-    const ownedItemHashes = new Set<number>();
+  (defs, profileResponse, allItems) => {
+    const accountWideOwned = new Set<number>();
+    const storeSpecificOwned = new Map<string, Set<number>>();
     for (const item of allItems) {
-      ownedItemHashes.add(item.hash);
+      if (STORE_SPECIFIC_OWNERSHIP_BUCKETS.includes(item.bucket.hash)) {
+        if (!storeSpecificOwned.has(item.owner)) {
+          storeSpecificOwned.set(item.owner, new Set());
+        }
+        storeSpecificOwned.get(item.owner)!.add(item.hash);
+      } else {
+        accountWideOwned.add(item.hash);
+      }
     }
-    if (profileResponse?.profilePlugSets?.data) {
-      for (const plugSet of Object.values(profileResponse.profilePlugSets.data.plugs)) {
-        for (const plug of plugSet) {
-          if (plug.canInsert) {
-            ownedItemHashes.add(plug.plugItemHash);
+    if (defs) {
+      const processPlugSet = (
+        plugs: { [key: number]: DestinyItemPlug[] },
+        insertInto: Set<number>
+      ) => {
+        const items = new Set<number>();
+        _.forIn(plugs, (plugs, plugSetHash) => {
+          const hash = parseInt(plugSetHash, 10);
+          if (universalOrnamentPlugSetHashes.includes(hash)) {
+            // #7597: Plugs from the universal ornament sets are exactly the armor items,
+            // so without this universal ornaments cause the corresponding armor items
+            // as being owned.
+            return;
+          }
+          const plugSetDef = defs.PlugSet.get(hash);
+          for (const item of plugSetDef.reusablePlugItems) {
+            const itemDef = defs.InventoryItem.get(item.plugItemHash);
+            if (plugs.some((k) => k.plugItemHash === itemDef.hash && k.enabled)) {
+              insertInto.add(itemDef.hash);
+            }
```

Comment:
I moved this code from `Vendors.tsx` but am not sure what it's trying to accomplish:

1. Why is it requesting the item from defs? Shouldn't `defs.InventoryItem.get(x).hash === x`?
2. Why is it looking at the PlugSet? Is there a case where the profile response lists plugs not present in the PlugSet that must be removed?

Reply:
https://github.com/DestinyItemManager/DIM/pull/4449

Reply:
Yeah that doesn't make a ton of sense to me either. I suspect it was leftover from when I was logging the whole item to try and understand it.

Reply:
I also agree, there doesn't seem to be a need to re-load the PlugSet.